Add doc and python script for pure thermal diffusion example (#2249)
GEOSX/GEOSX: src/docs/sphinx/advancedExamples/validationStudies/wellboreProblems/pureThermalDiffusion/Example.rst

.. _AdvancedExamplePureThermalDiffusionWellbore:

####################################################
Pure thermal diffusion around a wellbore
####################################################

------------------------------------------------------------------
Problem description
------------------------------------------------------------------

This example uses the thermal flow solver to a pure thermal diffusion problem around a wellbore. Thermal convection and fluid flow are removed by settting fluid pressure and fluid heat capacity to zero to mimic a pure thermal diffusion problem, .

> 

Reviewer: Adding a figure here could help to describe the settings of the wellbore problem.

Author: done

Analytical results of the temperature profile along the radial directionis `(Wang and Papamichos, 1994) <https://agupubs.onlinelibrary.wiley.com/doi/abs/10.1029/94WR01774>`__ :

.. math::
   T(r) = T_{in}\sqrt{\frac{R_{in}}{r}}erfc(\frac{r-R_{in}}{2\sqrt{c_{T}t}})

where :math:`r` is the radial coordinate, :math:`T_{in}` the temperature applied on the surface of the wellbore at :math:`r = R_{in}`, :math:`c_{T}` the thermal diffusion coefficient of rock that is the ratio between the thermal conductivity and ther volumetric heat capacity of rock.

**Input file**

This benchmark example uses no external input files and everything required is
contained within two GEOSX xml files that are located at:

.. code-block:: console

  inputFiles/singlePhaseFlow/thermalCompressible_2d_base.xml

and

.. code-block:: console

  inputFiles/singlePhaseFlow/thermalCompressible_2d_benchmark.xml

> The corresponding integrated test is

Reviewer: Smoke test is not found in this PR and would be added into integratedTests separately?

Author: It is already added before this PR. The xml file defining the smoke test is inputFiles/singlePhaseFlow/thermalCompressible_2d_smoke.xml

  :language: xml
  :start-after: <!-- SPHINX_SinglePhaseConstantThermalConductivity -->
  :end-before: <!-- SPHINX_SinglePhaseConstantThermalConductivityEnd -->

The volumetric heat capacity of fluid is set to a negligible value to minimize thermal convection effec. It is defined in the ``ThermalCompressibleSinglePhaseFluid`` XML block as  

.. literalinclude:: ../../../../../../../inputFiles/singlePhaseFlow/thermalCompressible_2d_base.xml
  :language: xml
  :start-after: <!-- SPHINX_ThermalCompressibleSinglePhaseFluid -->
  :end-before: <!-- SPHINX_ThermalCompressibleSinglePhaseFluidEnd -->

--------------------------------------------------------------------   
FieldSpecifications
--------------------------------------------------------------------

The initial temperature, the imposed temperature at the curved wellbore surface as well as the far-field temperature are defined using ``faceManager`` as

.. literalinclude:: ../../../../../../../inputFiles/singlePhaseFlow/thermalCompressible_2d_base.xml
  :language: xml
  :start-after: <!-- SPHINX_FieldSpecificationImposedTemperature -->
  :end-before: <!-- SPHINX_FieldSpecificationImposedTemperatureEnd -->

> Although a pure thermal diffusion problem is considered, it is required to define also specifications for fluid pressure because thermal transfer problem is always coupled with fluid flow problem in GEOSX. In this example, fluid pressure is set to zero everywhere to mimic a pure thermal diffusion problem as

Reviewer: ```suggestion
Although a pure thermal diffusion problem is considered, it is also required to define specifications for fluid pressure, as thermal transfer is always coupled with fluid flow in GEOSX. In this example, fluid pressure is set to zero everywhere to mimic a pure thermal diffusion problem as
```

.. _AdvancedExamplePureThermalDiffusionWellbore:

####################################################

> Pure thermal diffusion around a wellbore

Reviewer 2: ```suggestion
Pure Thermal Diffusion Around a Wellbore
```
to be consistent with the other titles

####################################################

------------------------------------------------------------------
Problem description
------------------------------------------------------------------

> This example uses the thermal flow solver to model a pure thermal diffusion problem around a wellbore. To mimic this specific problem, thermal convection and fluid flow are removed by setting fluid pressure and fluid heat capacity to zero. With a uniform temperature applied on the inner surface of the wellbore, temperature field would radially diffuse as shown in the figure below:

Reviewer 2: ```suggestion
This example uses the thermal single-phase flow solver to model a pure thermal diffusion problem around a wellbore. To mimic this specific problem, thermal convection and fluid flow are neglected by setting fluid pressure and fluid heat capacity to zero. With a uniform temperature applied on the inner surface of the wellbore, temperature field would radially diffuse as shown in the figure below:
```

  :language: xml
  :start-after: <!-- SPHINX_SolidInternalEnergy -->
  :end-before: <!-- SPHINX_SolidInternalEnergyEnd -->

The thermal conductivity of the medium around the wellbore is defined in the ``SinglePhaseConstantThermalConductivity`` XML block as  

.. literalinclude:: ../../../../../../../inputFiles/singlePhaseFlow/thermalCompressible_2d_base.xml
  :language: xml
  :start-after: <!-- SPHINX_SinglePhaseConstantThermalConductivity -->
  :end-before: <!-- SPHINX_SinglePhaseConstantThermalConductivityEnd -->

The volumetric heat capacity of fluid is set to a negligible value to exclude thermal convection effect. It is defined in the ``ThermalCompressibleSinglePhaseFluid`` XML block as  

.. literalinclude:: ../../../../../../../inputFiles/singlePhaseFlow/thermalCompressible_2d_base.xml
  :language: xml
  :start-after: <!-- SPHINX_ThermalCompressibleSinglePhaseFluid -->
  :end-before: <!-- SPHINX_ThermalCompressibleSinglePhaseFluidEnd -->

--------------------------------------------------------------------   
FieldSpecifications
--------------------------------------------------------------------

> The initial temperature, the imposed temperature at the curved wellbore surface as well as the far-field temperature are defined using ``faceManager`` as

Reviewer 2: ```suggestion
The initial temperature, the imposed temperature at the curved wellbore surface as well as the far-field temperature are defined as Dirichlet face boundary conditions using ``faceManager`` as
```
